cppcache/include/geode/Cache.hpp

```diff
@@ -76,7 +78,8 @@ class APACHE_GEODE_EXPORT Cache : public GeodeCache {
    * @param regionShortcut
    *        To create the region specific type, @see RegionShortcut
    */
-  virtual RegionFactory createRegionFactory(RegionShortcut regionShortcut);
```

Comment:
Region shortcut is an enum type. The size of the ref is actually more expensive than the enum value.

Reply:
That’s fine, it’s a small enum. Though I would expect a forward to work. See docs for `enum class`.

Reply:
You are right - forward declaration does work, CLion was giving me a false error, but it compiles fine.  I'll watch out for that sort of thing in the future.